sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ContinuousAccessEvaluationTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Security.KeyVault.Tests;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Administration.Tests
+{
+    [NonParallelizable]
+    internal class ContinuousAccessEvaluationTests : ContinuousAccessEvaluationTestsBase
+    {
+        [SetUp]
+        public void Setup()
+        {
+            ChallengeBasedAuthenticationPolicy.ClearCache();
+        }
+
+        [Test]
+        [TestCase(@"Bearer realm="""", authorization_uri=""https://login.microsoftonline.com/common/oauth2/authorize"", error=""insufficient_claims"", claims=""eyJhY2Nlc3NfdG9rZW4iOnsibmJmIjp7ImVzc2VudGlhbCI6dHJ1ZSwidmFsdWUiOiIxNzI2MDc3NTk1In0sInhtc19jYWVlcnJvciI6eyJ2YWx1ZSI6IjEwMDEyIn19fQ==""", """{"access_token":{"nbf":{"essential":true,"value":"1726077595"},"xms_caeerror":{"value":"10012"}}}""")]
+        public async Task VerifyCaeClaims(string challenge, string expectedClaims)
+        {
+            int callCount = 0;
+
+            MockResponse response = new MockResponse(200);
+
+            MockTransport transport = GetMockTransportWithCaeChallenges(numberOfCaeChallenges: 1, final200response: response);
+
+            var credential = new TokenCredentialStub((r, c) =>
+            {
+                if (callCount == 0)
+                {
+                    // The first challenge should not have any claims.
+                    Assert.IsNull(r.Claims);
+                }
+                else if (callCount == 1)
+                {
+                    Assert.AreEqual(expectedClaims, r.Claims);
+                }
+                Interlocked.Increment(ref callCount);
+                Assert.AreEqual(true, r.IsCaeEnabled);
+
+                return new(callCount.ToString(), DateTimeOffset.Now.AddHours(2));
+            }, true);
+
+            KeyVaultBackupClient client = new(
+            VaultUri,
+            credential,
+            new KeyVaultAdministrationClientOptions()
+            {
+                Transport = transport,
+            });
+
+            try
+            {
+                KeyVaultBackupOperation operation = await client.StartBackupAsync(VaultUri);
+            }
+            catch (RequestFailedException ex)
+            {
+                Assert.AreEqual(200, ex.Status);
```

Comment:
This seems to imply we expect the client to throw RequestFailedException after getting a 200 response. Is that correct?

Reply:
That’s correct. To keep these tests simple, I’m using a dummy 200 MockResponse without any content. The focus of this test is on the claims.

Reply:
But why would the client throw? A 200 response indicates success

Reply:
It throws because the `MockResponse` is incomplete, lacking both `headers` and `contentStream`, making it invalid for the client request. Emulating a real response for Admin was challenging, so I went with this simplified version to proceed with the release. The response code is irrelevant for this test, since I’m just verifying that it matches the one specified in the mock transport.